### ISSUE TITLE
`uname -p` check reports the cpu model name on WSL

### ISSUE THREAD

**Issue by dstanberry** [ellisonleao/glow.nvim#58]
https://github.com/ellisonleao/glow.nvim/blob/d86281307ce2898d0fcd85ecb0865fc1dd2f2578/lua/glow.lua#L88-L95

I've checked with two wsl distros and for me `uname -p` reports the cpu model (not x86_64 for example).
`uname -m` does however respond with x86_64. Due to this the order of assignment for `arch` the utility fails incorrectly. Why is the `-p` needed at all?

**Comment by nrj5k** [comment]
I use Slackware and I faced a similar issue too. Fixed it by changing `-p` to `-m` and GlowInstall worked

**Comment by ellisonleao** [comment]
does `uname -m` works for all linux dists ? we can work on a new strategy to get this value for sure. If you guys have another option just create a new PR or put some comments here and we can discuss it. Thanks!

**Comment by nrj5k** [comment]
Maybe it is something between uname versions? I have version 9.0,

It might be easier to use `uname -a` and do a string search for the arch. `uname -a | grep -wo <arch>` should be a valid replacement, but you'll have to iterate over the archs.

**Comment by JuanCldCmt** [comment]
Is this bug still here after #73 ? I've checked both GNU coreutils and BSDs' man pages:
```
GNU:
-m, --machine
              print the machine hardware name
BSD:
-m	     Write the type of the current hardware platform to	standard out-
	     put.  ([make(1)](https://www.freebsd.org/cgi/man.cgi?query=make&sektion=1&apropos=0&manpath=FreeBSD+13.1-RELEASE+and+Ports) uses it to set the MACHINE variable.)
```
Maybe on macOS, the `uname -p` returns the architecture.
I don't have a BSD system for testing, but I think if you are using GNU's coreutils (Which is what most linux users would use), 
`uname -m` should return the right answer.
and this bug should be fixed after #73 .

**Comment by dstanberry** [comment]
I agree #73 seems to resolve this. Closing now..
